docs/hazmat/primitives/asymmetric/ec.rst

```diff
@@ -122,6 +122,43 @@ Elliptic Curve Signature Algorithms
         :returns: A new instance of a :class:`EllipticCurvePublicKey`
             provider.
 
+Elliptic Curve Key Exchange algorithm
+-------------------------------------
+
+.. class:: ECDH(algorithm)
```

Comment:
So to be clear, `get_public` (in the API proposals @ronf provided this morning) is the spiritual equivalent of `exchange` and `get_shared` is `agree`?

Our current API has `exchange` added to the private key object, but I like the new one better.

```python
private_key = ec.generate_private_key(ec.SECP384R1(), default_backend())
ecdh = ECDH(private_key)
public_key = ecdh.get_public()
shared_key = ecdh.get_shared(peer_public_key, KDF())
```
instead of
```python
private_key = ec.generate_private_key(ec.SECP384R1(), default_backend())
keyex = private_key.exchange(ec.ECDH(KDF()))
sharedkey = keyex.agree(public_key)
```
with a future PR to add ECDHE like
```python
ecdhe = ECDHe(ec.SECP256R1(), default_backend())
public_key = ecdhe.get_public()
shared_key = ecdhe.get_shared(peer_public_key, KDF())
```

cryptography's APIs have strongly leaned toward explicit over implicit, so I'm in favor of requiring `get_shared` to have a KDF argument. As @simo5 said, you can pass a NullKDF for your purposes to obtain the key unmolested, but it seems prudent to strongly encourage the use of KDF by design (many consumers of this will inevitably end up using it outside the context of implementing an existing protocol like SSH).